tests/ioc/test_file_server.py

```diff
@@ -0,0 +1,65 @@
+import asyncio
+import json
+import os
+from typing import Dict
+
+import aiohttp
+import pytest
+from aiohttp.test_utils import TestClient, TestServer
+
+from pandablocks.ioc.ioc import initialise_server
+from tests.conftest import DummyServer
+
+# Tests for the bobfile server
+
+test_bob_file_dict: Dict[str, str] = {"TEST.bob": "<Bobfile>Contents</Bobfile>"}
+
+
+@pytest.fixture
+async def setup_server():
+    """Adds the test server to the current event loop and creates a test client."""
+    loop = asyncio.get_event_loop()
+    app = initialise_server(test_bob_file_dict)
+    async with TestClient(TestServer(app), loop=loop) as client:
+        yield client
+
+
+@pytest.mark.asyncio
+async def test_get_available_files(setup_server):
+    """Tests a request for the available files."""
+    client = setup_server
+    resp = await client.get("/")
+    assert resp.status == 200
+    text = await resp.text()
+    assert text == '["TEST.bob"]'
+
+
+@pytest.mark.asyncio
+async def test_get_file(setup_server):
+    """Tests a request for a single file."""
+    client = setup_server
+    resp = await client.get("/TEST.bob")
+    assert resp.status == 200
+    text = await resp.text()
+    assert text == test_bob_file_dict["TEST.bob"]
+
+
+@pytest.mark.asyncio
+async def test_system_bobfile_creation(
+    dummy_server_system: DummyServer, subprocess_ioc
+):
+    """A system test for both the bobfile creation and running the server."""
+    loop = asyncio.get_event_loop()
+    async with aiohttp.ClientSession(loop=loop) as session:
+
+        async with session.get("http://0.0.0.0:8080/") as response:
```

Comment:
I'm thinking that the host and port numbers should be defined as constants in `ioc.py` and imported here. I say that as I don't think we really care what the defaults are, so there's little point in testing them.

An alternative is to have another test that does hard-code these values, so that if they ever do change there's a test failure the coder has to investigate. The other tests can still use imported values.